qt/python/mantidqt/mantidqt/widgets/helpwindow/helpwindowpresenter.py

```diff
@@ -4,56 +4,97 @@
+        """
+        if self._view is None:
+            logger.debug("Creating HelpWindowView instance.")
+            interceptor = self.model.create_request_interceptor()
+            modeString = self.model.get_mode_string()
+            isLocal = self.model.is_local_docs_mode()
+            self._view = HelpWindowView(self, interceptor=interceptor)
+            self._view.set_status_indicator(modeString, isLocal)
+            logger.debug("HelpWindowView instance created.")
 
-    def showHelpPage(self, relative_url):
+    def show_help_page(self, relativeUrl):
         """
         Build the doc URL from the Model and tell the View to load it.
+        Ensures the View is created and visible.
         """
-        doc_url = self.model.build_help_url(relative_url)
-        self.view.set_page_url(doc_url)
+        self._ensure_view_created()
+        logger.debug(f"Requesting help page: '{relativeUrl}'")
+        docUrl = self.model.build_help_url(relativeUrl)
+        self._view.set_page_url(docUrl)
         self.show_help_window()
 
     def show_home_page(self):
         """
-        Presenter logic to load the 'Home' page, which might be local or online.
+        Presenter loggeric to load the 'Home' page, which might be local or online.
         The View calls this when the user hits the Home button.
         """
-        home_url = self.model.get_home_url()
-        self.view.set_page_url(home_url)
-        self.show_help_window()
+        self._ensure_view_created()
+        logger.debug("Requesting home page.")
+        homeUrl = self.model.get_home_url()
+        self._view.set_page_url(homeUrl)
 
     def show_help_window(self):
-        if not self._window_open:
-            self._window_open = True
-            self.view.display()
+        self._ensure_view_created()
+        if not self._windowOpen:
+            logger.debug("Displaying help window for the first time.")
+            self._windowOpen = True
+            self._view.display()
+        else:
+            logger.debug("Raising existing help window.")
+            self._view.show()
+            self._view.raise_()
+            self._view.activateWindow()
 
     def on_close(self):
-        """Called by the View when the window closes."""
-        print("Help window closed.")
-        self.cleanup()
+        """
+        Called by the View when the window closes.
+        """
+        logger.debug("View signaled interactive close.")
+        self._windowOpen = False
 
     def cleanup(self):
-        """Cleanup resources, close the View if open."""
-        if self._window_open:
-            self._window_open = False
-            print("Cleaning up Help Window resources.")
-            self.view.close()
+        """
+        Cleanup resources, close the View if open. Called on app quit.
+        """
+        logger.debug("Cleanup requested (likely app quitting).")
+        if self._view is not None and self._windowOpen:
```

Comment:
This is a very negative if/else tree. With less branches ruff would have seen it and complained more. My suggestion is
```python
if self._view:
    if self._windowOpen:
        # do stuff
    else:
        # do other stuff
    self._view.close()
    self._view = None
else:
    logger.debug("No view instance to clean up.")
```
In python, `bool(None) == False` so you shouldn't need to directly check against `None`. The alternative is to flip the two options and check for `if self._view is None:` at the top